File: src/Entity/Evaluation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Entity\Projets;
use Doctrine\Common\Collections\Collection;
use App\Entity\Vote;

#[ORM\Entity]
class Evaluation
{

    #[ORM\Id]
    #[ORM\Column(type: "integer")]
    private int $id;

        #[ORM\ManyToOne(targetEntity: User::class, inversedBy: "evaluations")]
    #[ORM\JoinColumn(name: 'idJury', referencedColumnName: 'id_user', onDelete: 'CASCADE')]
    private User $idJury;

        #[ORM\ManyToOne(targetEntity: Hackathon::class, inversedBy: "evaluations")]
    #[ORM\JoinColumn(name: 'idHackathon', referencedColumnName: 'id_hackathon', onDelete: 'CASCADE')]
    private Hackathon $idHackathon;

        #[ORM\ManyToOne(targetEntity: Projets::class, inversedBy: "evaluations")]
    #[ORM\JoinColumn(name: 'idProjet', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Projets $idProjet;

    #[ORM\Column(type: "float")]
    private float $NoteTech;

    #[ORM\Column(type: "float")]
    private float $NoteInnov;

    #[ORM\Column(type: "date")]
    private \DateTimeInterface $date;

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getIdJury()
    {
        return $this->idJury;
    }

    public function setIdJury($value)
    {
        $this->idJury = $value;
    }

    public function getIdHackathon()
    {
        return $this->idHackathon;
    }

    public function setIdHackathon($value)
    {
        $this->idHackathon = $value;
    }

    public function getIdProjet()
    {
        return $this->idProjet;
    }

    public function setIdProjet($value)
    {
        $this->idProjet = $value;
    }

    public function getNoteTech()
    {
        return $this->NoteTech;
    }

    public function setNoteTech($value)
    {
        $this->NoteTech = $value;
    }

    public function getNoteInnov()
    {
        return $this->NoteInnov;
    }

    public function setNoteInnov($value)
    {
        $this->NoteInnov = $value;
    }

    public function getDate()
    {
        return $this->date;
    }

    public function setDate($value)
    {
        $this->date = $value;
    }

    #[ORM\OneToMany(mappedBy: "idEvaluation", targetEntity: Vote::class)]
    private Collection $votes;

        public function getVotes(): Collection
        {
            return $this->votes;
        }
    
        public function addVote(Vote $vote): self
        {
            if (!$this->votes->contains($vote)) {
                $this->votes[] = $vote;
                $vote->setIdEvaluation($this);
            }
    
            return $this;
        }
    
        public function removeVote(Vote $vote): self
        {
            if ($this->votes->removeElement($vote)) {
                // set the owning side to null (unless already changed)
                if ($vote->getIdEvaluation() === $this) {
                    $vote->setIdEvaluation(null);
                }
            }
    
            return $this;
        }
}

File: OneDrive/Bureau/Hackify-integrationn/src/main/java/models/Evaluation.java


package models;

public class Evaluation {
    int id;
    int idJury;
    int idProjet;
    int idHackathon;
    float noteTech;
    float noteInnov;
    String date;

    public Evaluation() {
    }

    public Evaluation(int id, int idJury, int idHackathon, int idProjet, float noteTech, float noteInnov, String date) {
        this.id = id;
        this.noteTech = noteTech;
        this.date = date;
        this.noteInnov = noteInnov;
        this.idJury = idJury;
        this.idProjet = idProjet;
        this.noteInnov = noteInnov;
    }

    public Evaluation(float noteTech, float noteInnov, String date, int idJury, int idProjet, int idHackathon) {
        this.noteTech = noteTech;
        this.noteInnov = noteInnov;
        this.date = date;
        this.idJury = idJury;
        this.idProjet = idProjet;
        this.idHackathon = idHackathon;
    }

    public int getIdHackathon() {
        return this.idHackathon;
    }

    public void setIdHackathon(int idHackathon) {
        this.idHackathon = idHackathon;
    }

    public int getIdJury() {
        return this.idJury;
    }

    public void setIdJury(int idJury) {
        this.idJury = idJury;
    }

    public int getIdProjet() {
        return this.idProjet;
    }

    public void setIdProjet(int idProjet) {
        this.idProjet = idProjet;
    }

    public int getId() {
        return this.id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public float getNoteInnov() {
        return this.noteInnov;
    }

    public void setNoteInnov(float noteInnov) {
        this.noteInnov = noteInnov;
    }

    public float getNoteTech() {
        return this.noteTech;
    }

    public void setNoteTech(float noteTech) {
        this.noteTech = noteTech;
    }

    public String getDate() {
        return this.date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String toString() {
        return "id=" + this.id + ", idJury=" + this.idJury + ", idProjet=" + this.idProjet + ", idHackathon=" + this.idHackathon + ", noteTech=" + this.noteTech + ", noteInnov=" + this.noteInnov + ", date='" + this.date;
    }
}
